packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics-latencies.ts

```diff
@@ -0,0 +1,290 @@
+  }
+
+  /**
+   * Click To Interstitial
+   * @returns - latency
+   */
+  public getClickToInterstitial() {
+    return this.getDiffBetweenTimestamps(
+      'client.meeting.click.joinbutton',
+      'client.meeting.interstitial-window.showed'
+    );
+  }
+
+  /**
+   * Interstitial To Join Ok
+   * @returns - latency
+   */
+  public getInterstitialToJoinOK() {
+    return this.getDiffBetweenTimestamps(
+      'client.meeting.click.joinbutton',
+      'client.locus.join.response'
+    );
+  }
+
+  /**
+   * Interstitial To Media Ok
+   * @returns - latency
+   */
+  public getInterstitialToMediaOK() {
+    return this.getDiffBetweenTimestamps(
+      'client.meeting.click.joinbutton',
+      'sdk.media-flow.started'
+    );
+  }
+
+  /**
+   * Total JMT
+   * @returns - latency
+   */
+  public getTotalJMT() {
+    const clickToInterstitial = this.getClickToInterstitial();
+    const interstitialToJoinOk = this.getInterstitialToJoinOK();
+
+    if (clickToInterstitial && interstitialToJoinOk) {
+      return clickToInterstitial + interstitialToJoinOk;
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Join Conf JMT
+   * @returns - latency
+   */
+  public getJoinConfJMT() {
+    const joinReqResp = this.getJoinReqResp();
+    const ICESetupTime = this.getICESetupTime();
+
+    if (joinReqResp && ICESetupTime) {
+      return joinReqResp + ICESetupTime;
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Total Media JMT
+   * @returns - latency
+   */
+  public getTotalMediaJMT() {
+    const clickToInterstitial = this.getClickToInterstitial();
+    const interstitialToJoinOk = this.getInterstitialToJoinOK();
+    const joinConfJMT = this.getJoinConfJMT();
+
+    if (clickToInterstitial && interstitialToJoinOk && joinConfJMT) {
+      return clickToInterstitial + interstitialToJoinOk + joinConfJMT;
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Client JMT
+   * @returns - latency
+   */
+  public getClientJMT() {
+    const interstitialToJoinOk = this.getInterstitialToJoinOK();
+    const joinConfJMT = this.getJoinConfJMT();
+
+    if (interstitialToJoinOk && joinConfJMT) {
+      return interstitialToJoinOk + joinConfJMT;
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Audio setup delay receive
+   */
+  public getAudioJoinRespRxStart() {
+    return this.getDiffBetweenTimestamps('client.locus.join.response', 'client.media.rx.start');
+  }
+
+  /**
+   * Video setup delay receive
+   */
+  public getVideoJoinRespRxStart() {
+    return this.getDiffBetweenTimestamps('client.locus.join.response', 'client.media.rx.start');
+  }
+
+  /**
+   * Audio setup delay transmit
+   */
+  public getAudioJoinRespTxStart() {
+    return this.getDiffBetweenTimestamps('client.locus.join.response', 'client.media.tx.start');
+  }
+
+  /**
+   * Video setup delay transmit
+   */
+  public getVideoJoinRespTxStart() {
+    return this.getDiffBetweenTimestamps('client.locus.join.response', 'client.media.tx.start');
```

Comment:
this is the same as getAudioJoinRespTxStart() - just double checking that that's on purpose? same for getVideoJoinRespRxStart and getAudioJoinRespRxStart